jupyter_publishing_service/authorizer/sqlrbac.py

```diff
@@ -40,7 +43,13 @@ async def authorize(self, user, data) -> bool:
             user_permissions = results.all()
             if all(perms in user_permissions for perms in required_perms):
                 return True
+
+            c_stmt = select(SharedFileMetadata.id).where(CollaboratorRole.id == file_id)
+            results = await session.exec(c_stmt)
+            item_missing = results.one_or_none() is None
+            if item_missing:
+                raise HTTPException(status_code=404, detail="The file ID requested does not exist.")
```

Comment:
Wanted to avoid this check as the system would be checking for a non-existent file for a spurious request. This may lead to exhaustion of server resources and a target for DOS

Reply:
Hm, I see. Let me think about this some more. 

I just found the error very confusing when a file wasn't found. I wonder if there is something better we can do here.